pytorch_lightning/utilities/argparse.py

```diff
@@ -15,12 +15,13 @@
 import os
 from argparse import _ArgumentGroup, ArgumentParser, Namespace
 from contextlib import suppress
-from typing import Any, Dict, List, Tuple, Union
+from typing import Any, Callable, Dict, List, Tuple, Type, Union
 
+import pytorch_lightning as pl
 from pytorch_lightning.utilities.parsing import str_to_bool, str_to_bool_or_int, str_to_bool_or_str
 
 
-def from_argparse_args(cls, args: Union[Namespace, ArgumentParser], **kwargs):
+def from_argparse_args(cls: Type['pl.Trainer'], args: Union[Namespace, ArgumentParser], **kwargs: Any) -> 'pl.Trainer':
```

Comment:
All these other instances that use 'pl.Trainer' should also be updated. Probably with a `TypeVar` like in this case:

```suggestion
def from_argparse_args(cls: Type[T], args: Union[Namespace, ArgumentParser], **kwargs: Any) -> T:
```